python/lsst/meas/extensions/trailedSources/NaivePlugin.py

```diff
@@ -191,3 +201,61 @@ def fail(self, measRecord, error=None):
             self.flagHandler.handleFailure(measRecord)
         else:
             self.flagHandler.handleFailure(measRecord, error.cpp)
+
+    def computeSecondMomentDiff(self, z, c):
+        """Compute difference of the numerical and analytic second moments.
+
+        Parameters
+        ----------
+        z : `float`
+            Proportional to the length of the trail. (see notes)
+        c : `float`
+            Constant (see notes)
+
+        Returns
+        -------
+        diff : `float`
+            Difference in numerical and analytic second moments.
+
+        Notes
+        -----
+        This is a simplified expression for the difference between the stack
+        computed adaptive second-moment and the analytic solution. The variable
+        z is proportional to the length such that L = 2*z*sqrt(2*(Ixx+Iyy)),
+        and c is a constant (c = 4*Ixx/((Ixx+Iyy)*sqrt(pi))). Both have been
+        defined to avoid unnecessary floating-point operations in the root
+        finder.
+        """
+
+        diff = erf(z) - c*z*np.exp(-z*z)
+        return diff
+
+    def findLength(self, Ixx, Iyy):
+        """Find the length of a trail, given adaptive second-moments.
+
+        Uses a root finder to compute the length of a trail corresponding to
+        the adaptive second-moments computed by previous measurements
+        (ie. SdssShape).
+
+        Parameters
+        ----------
+        Ixx : `float`
+            Adaptive second-moment along x-axis.
+        Iyy : `float`
+            Adaptive second-moment along y-axis.
+
+        Returns
+        -------
+        length : `float`
+            Length of the trail.
+        results : `scipy.optimize.RootResults`
+            Contains messages about convergence from the root finder.
+        """
+
+        xpy = Ixx + Iyy
+        c = 4.0*Ixx/(xpy*np.sqrt(np.pi))
+        z, results = sciOpt.brentq(lambda z: self.computeSecondMomentDiff(z, c),
+                                   0.01, 1.0, full_output=True)
```

Comment:
magic numbers that have no configuration should (mostly) be avoided. Either justify 0.01 and 1 here, or better, add config fields so these numbers can be tweaked without editing the code if need be, and use these numbers as the default value.

Reply:
Ok, I've added some comments to (hopefully) clarify. In this case, having static values makes sense. The roots are bounded, based on the allowed values of c.